hawc/apps/udf/views.py

```diff
@@ -1,9 +1,31 @@
-from django.urls import reverse
-from django.views.generic.edit import CreateView
+from django.urls import reverse_lazy
+from django.utils.decorators import method_decorator
+from django.views.generic import DetailView, ListView
+from django.views.generic.edit import CreateView, FormView, UpdateView
 
-from hawc.apps.common.views import LoginRequiredMixin, MessageMixin
+from hawc.apps.common import dynamic_forms
+from hawc.apps.common.views import LoginRequiredMixin, MessageMixin, htmx_required
 
-from .forms import UDFForm
+from . import models
+from .forms import SchemaPreviewForm, UDFForm
+
+
+class UDFListView(LoginRequiredMixin, ListView):
+    template_name = "udf/udf_list.html"
+    model = models.UserDefinedForm
+
+
+class UDFDetailView(LoginRequiredMixin, DetailView):
+    template_name = "udf/udf_detail.html"
+    model = models.UserDefinedForm
+
+    def get_context_data(self, **kwargs):
+        form = dynamic_forms.Schema.parse_obj(self.object.schema).to_form()
+        kwargs.update(form=form)
+        return super().get_context_data(**kwargs)
+
+    def user_can_edit(self):
```

Comment:
I didn't know that `view` is available in a template context, thanks for showing me this!

Reply:
I happened to notice it when looking at the context in the debug menu for the template, it turned out to be really useful here!